sklearn/utils/estimator_checks.py

```diff
@@ -2120,6 +2123,119 @@ def check_classifiers_multilabel_representation_invariance(
     assert type(y_pred) == type(y_pred_list_of_lists)
 
 
+@ignore_warnings(category=FutureWarning)
+def check_classifiers_multilabel_format_output(name, classifier_orig):
+    """Check the output of the response methods for classifiers supporting
+    multilabel-indicator targets."""
+    classifier = clone(classifier_orig)
+    set_random_state(classifier)
+
+    n_samples, test_size, n_outputs = 100, 25, 5
+    X, y = make_multilabel_classification(n_samples=n_samples, n_features=2,
+                                          n_classes=n_outputs, n_labels=3,
+                                          length=50, allow_unlabeled=True,
+                                          random_state=0)
+    X = scale(X)
+
+    X_train, X_test = X[:-test_size], X[-test_size:]
+    y_train, y_test = y[:-test_size], y[-test_size:]
+    classifier.fit(X_train, y_train)
+
+    response_method_name = ["predict", "predict_proba", "decision_function"]
+    for method_name in response_method_name:
+        response_method = getattr(classifier, method_name, None)
+        if response_method is not None:
+            y_pred = response_method(X_test)
+
+            if method_name == "predict":
```

Comment:
Since `method_name` all has its own distinct path, I think we do not need the for loop? 

```
predict_method = getattr(classifier, "predict", None)
if predict_method is not None:
    ...

predict_proba_method = ...
if predict_proba_method is not None:
    ...
```

This would also help with lower the indentations.

Reply:
OK. Be aware that this comment will be addressed in #19859 :)
I did not want to write specific test for this issue knowing that it should be treated in the common tests.